### ISSUE TITLE
Coupon codes should be case-insensitive, but they're not on PostgreSQL

### ISSUE THREAD

**Issue by boboldehampsink** [craftcms/commerce#2249]
**Description**
Coupon codes should be case-insensitive, but they're not on PostgreSQL.

Problem is this line:

```php
$discounts = $this->_createDiscountQuery()->andWhere(['[[discounts.code]]' => $code])->all();
```

it's case sensitive...  see https://stackoverflow.com/questions/7005302/postgresql-how-to-make-case-insensitive-query for solutions

**To Reproduce**
How, and what happened?
1. Create Coupon code
2. Try to submit it with different casing

**Expected behavior**
Should work, but doesn't.

**Additional info**
- Craft version: latest
- Commerce version: latest
- PHP version: 8.0.8
- Database driver & version: PostgreSQL
- Other Plugins & versions:

**Additional context**
Probably related to changes in https://github.com/craftcms/commerce/commit/df0a71907802665c1702b2c04f25087ff13f057e

**Comment by pdaleramirez** [comment]
@boboldehampsink I've tested coupon code case sensitivity on Postgres database and wasn't able to encounter any issue. I still get valid discount despite of the case.

I checked the code and the matching of the coupon code case sensitivity is via PHP `strcasecmp` so this couldn't be a case of DB matching.

**Comment by boboldehampsink** [comment]
@pdaleramirez the `strcasecmp` is only executed after the db query, which already yields no results because of case-sensitivity in postgres. What version of Postgres are you testing on?

**Comment by boboldehampsink** [comment]
@pdaleramirez this was an actual issue that has been fixed in #2252

**Comment by pdaleramirez** [comment]
@boboldehampsink Thank you for reporting it
